src/paradigma/tremor/feature_extraction.py

```diff
@@ -0,0 +1,429 @@
+        bandpower = spectral_resolution*np.sum(psd_window[freq_idx])
+        low_freq_power.append(bandpower)
+
+    return low_freq_power
+
+def extract_tremor_power(
+    total_psd: pd.Series,
+    freq_vect: pd.Series,
+    fmin: float = 3,
+    fmax: float = 7,
+    spectral_resolution: float = 0.25
+    ) -> pd.Series:
+
+    """Computes the tremor power (1.25 Hz around the peak within the tremor frequency band)
+    
+    Parameters
+    ----------
+    total_psd: pd.Series
+        The total power spectral density of the gyroscope signal
+    freq_vect: pd.Series
+        Frequency vector corresponding to the power spectral density
+    fmin: float
+        The lower bound of the tremor frequency band in Hz (default: 3)
+    fmax: float
+        The upper bound of the tremor frequency band in Hz (default: 7)
+    spectral_resolution: float
+        The spectral resolution of the PSD in Hz (default: 0.25)
+        
+    Returns
+    -------
+    pd.Series
+        The tremor power across windows
+    """
+
+    tremor_power = []
+    
+    freq_range = freq_vect[0]
+    freq_idx = np.where((freq_range>=fmin) & (freq_range<=fmax))
+    for psd_window in total_psd: 
+        peak_idx = np.argmax(psd_window[freq_idx]) + np.min(freq_idx)
+        left_idx =  np.max([0,int(peak_idx - 0.5/spectral_resolution)])
+        right_idx = int(peak_idx + 0.5/spectral_resolution)
+        peak_power = spectral_resolution*np.sum(psd_window[left_idx:right_idx+1])
+        tremor_power.append(peak_power)
+
+    return tremor_power
+
+def extract_spectral_domain_features(config, df_windowed):
+
+    # transform the temporal signal to the spectral domain using Welch's method and short time fourier transform
+    for col in config.l_gyroscope_cols:
+        df_windowed[f'{col}_freqs_PSD'], df_windowed[f'{col}_PSD'] = signal_to_PSD(
+            sensor_col = df_windowed[col], 
+            sampling_frequency = config.sampling_frequency,
+            window_type = config.window_type, 
+            segment_length_s = config.segment_length_s_psd,
+            overlap_fraction = config.overlap_fraction,
+            spectral_resolution = config.spectral_resolution_psd
+            )
+        df_windowed[f'{col}_spectrogram'] = signal_to_spectrogram(
+            sensor_col = df_windowed[col], 
+            sampling_frequency = config.sampling_frequency,
+            window_type = config.window_type, 
+            segment_length_s = config.segment_length_s_mfcc,
+            overlap_fraction = config.overlap_fraction
+            )
+    
+    # compute the total PSD and spectrogram (summed across the 3 gyroscope axes)
+    df_windowed['total_PSD'] = df_windowed[[f"{y}_PSD" for y in config.l_gyroscope_cols]].sum(axis=1)
+    df_windowed['total_spectrogram'] = df_windowed[[f"{y}_spectrogram"for y in config.l_gyroscope_cols]].sum(axis=1) 
+
+    # compute the cepstral coefficients
+    mfcc_cols = generate_mel_frequency_cepstral_coefficients(
```

Comment:
Zijn de waarden nu hetzelfde als in Matlab? Anders kunnen we hier volgende week maandag even voor zitten misschien?